include/cuco/detail/static_set/static_set_ref.inl

```diff
@@ -210,6 +210,45 @@ auto static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators..
     std::move(*this)};
 }
 
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Operators>
+template <typename NewKeyEqual>
+__host__ __device__ constexpr auto
+static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>::with_key_eq(
+  NewKeyEqual const& key_equal) const noexcept
+{
+  return static_set_ref<Key, Scope, NewKeyEqual, ProbingScheme, StorageRef, Operators...>{
+    cuco::empty_key<Key>{this->empty_key_sentinel()},
+    key_equal,
```

Comment:
Would it make sense to `std::forward` these parameters?

Reply:
Our ctor arguments are either a value of a const ref thus forwarding them seems not proper.